lib/rules/camelcase.js

```diff
@@ -11,6 +11,8 @@
 
 module.exports = {
     meta: {
+        type: "style",
```

Comment:
I'm still in the "style" camp on this one. Any change to resolve this issue requires changing the value of an AST node but does not change the structure of the program.

From an end-user point of view, I think I would expect a rule enforcing naming conventions to be listed as a style preference.

Reply:
I'd also lean towards `"suggestion"` here because in the long term, I think it would be valuable to be consistent about the idea that style rules don't modify the AST.

For example, if we add a way to disable all stylistic rules in the future, then someone using Prettier would be able to use this to turn off the rules that Prettier will handle. However, if we have exceptions where rules like `camelcase` are marked as stylistic despite modifying the AST, then a Prettier user would need to manually keep track of those rules and reenable them.

Reply:
Okay, seems like others are rallying around suggestion, so happy to go along.

One thing I want to make sure to call out: there is a difference between modifying the structure of the AST vs. modifying a value inside of an AST node. Changing the structure of the AST means changing the meaning of the program while modifying a value inside of an AST node means the meaning of the program does not change even though a symbol within the program does. 

If everyone is more comfortable with "style" meaning changing only those things that are not represented in the AST, I can go along with that.